packages/devtools_app/lib/src/inspector/flutter/story_of_your_layout/flex.dart

```diff
@@ -4,204 +4,549 @@
+            arrowStrokeWidth: 1.5,
+            child: RotatedBox(
+              quarterTurns: 1,
+              child: Text(
+                'height: ${size.height.toStringAsFixed(1)}',
+                textAlign: TextAlign.center,
+              ),
+            ),
+            direction: Axis.vertical,
           ),
-          distanceToArrow: 4.0,
-          type: ArrowType.right,
+          height: screenSize.height - kWidthArrowIndicatorHeight - kMargin,
+          margin: const EdgeInsets.only(left: kMargin),
         ),
-        margin: const EdgeInsets.only(bottom: 16.0),
-        width: length,
-      ),
-      left: Container(
-        child: ArrowWrapper.unidirectional(
-          arrowColor: theme.hintColor,
-          child: Text(
-            properties.verticalDirectionDescription,
-            textAlign: TextAlign.center,
+        rightWidth: kHeightArrowIndicatorWidth,
+        bottom: Container(
+          margin: const EdgeInsets.only(top: kMargin),
+          child: ArrowWrapper.bidirectional(
+            arrowColor: widthIndicatorColor,
+            arrowStrokeWidth: 1.5,
+            child: Text(
+              'width: ${size.width}',
+              textAlign: TextAlign.center,
+            ),
+            direction: Axis.horizontal,
           ),
-          distanceToArrow: 4.0,
-          type: ArrowType.down,
+          width: screenSize.width - kHeightArrowIndicatorWidth - kMargin,
         ),
-        height: length,
-        width: length * 0.25,
-      ),
-      right: Container(
-        child: ArrowWrapper.bidirectional(
-          arrowColor: theme.splashColor,
-          arrowStrokeWidth: 1.0,
-          child: Column(
-            children: <Widget>[
-              Text(
-                // TODO(albertusangga): Use real height
-                '${length.round()} px',
-                textAlign: TextAlign.center,
+        bottomHeight: kWidthArrowIndicatorHeight,
+      );
+    });
+  }
+
+  Widget _visualizeChild({
+    LayoutProperties node,
+    Color borderColor,
+    Color textColor,
+    double renderSmallestWidth,
+    double renderSmallestHeight,
+    double renderLargestWidth,
+    double renderLargestHeight,
+  }) {
+    final size = node.size;
+    final width = size.width;
+    final height = size.height;
+
+    final smallestWidth = properties.smallestWidthChild.size.width;
+    final smallestHeight = properties.smallestHeightChild.size.height;
+    final largestWidth = properties.largestWidthChild.size.width;
+    final largestHeight = properties.largestHeightChild.size.height;
+
+    final widthDifference =
+        largestWidth == smallestWidth ? 1 : (largestWidth - smallestWidth);
+    final heightDifference =
+        largestHeight == smallestHeight ? 1 : (largestHeight - smallestHeight);
+
+    final renderWidthDiff = renderLargestWidth - renderSmallestWidth;
+    final renderHeightDiff = renderLargestHeight - renderSmallestHeight;
+
+    final renderWidth =
```

Comment:
add a todo to refactor this code to reuse the same helper code to perform the calculations for width and height.